### PR TITLE
Add ResNeSt Segmentation Code

### PR DESCRIPTION
**Pull request by zhanghang1989**
I am retraining the models, will provide pretrained weights later.

### CODE REVIEW

**Comment by mli** [comment]
Job PR-1254-2 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1254/2/index.html 
Code coverage of this PR:  vs. Master:

File: scripts/segmentation/train.py [hunk 1]
```diff
@@ -158,12 +160,20 @@ def __init__(self, args, logger):
 
         # create network
         if args.model_zoo is not None:
-            model = get_model(args.model_zoo, pretrained=True)
+            model = get_model(args.model_zoo, norm_layer=args.norm_layer,
```
Comment (bryanyzhu) [inline scripts/segmentation/train.py:163]: Do we need to keep the `pretrained=True` since these are model_zoo models? Otherwise, the `get_model ` here will be just like `get_segmentation_model ` below.

File: scripts/segmentation/train.py [hunk 2]
```diff
@@ -184,10 +194,14 @@ def __init__(self, args, logger):
         self.criterion = DataParallelCriterion(criterion, args.ctx, args.syncbn)
 
         # optimizer and lr scheduling
-        self.lr_scheduler = LRScheduler(mode='poly', base_lr=args.lr,
-                                        nepochs=args.epochs,
-                                        iters_per_epoch=len(self.train_data),
-                                        power=0.9)
+        self.lr_scheduler = LRSequential([
+                LRScheduler('linear', base_lr=0, target_lr=args.lr,
```
[the remark this answers is not in the document]
Comment (zhanghang1989) [reply scripts/segmentation/train.py:198]: Yeah, I found ResNeSt-200 and 269 requires warm-up (otherwise diverge). I guess the issue of ResNeSt-101 on cityscapes has the same problem.

File: scripts/segmentation/train.py (hunk 1 above)
Comment (zhanghang1989) [reply scripts/segmentation/train.py:163]: Will add a `--pretrained` option.

**Comment by mli** [comment]
Job PR-1254-6 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1254/6/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1254-8 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1254/8/index.html 
Code coverage of this PR:  vs. Master:
